common/test/src/View/ResizeBrushesToolTest.cpp

```diff
@@ -72,5 +81,67 @@ namespace TrenchBroom {
+
+        /**
+         * Test for https://github.com/TrenchBroom/TrenchBroom/issues/3726
+         */
+        TEST_CASE("ResizeBrushesToolTest.findDragFaces", "[ResizeBrushesToolTest]") {
+            const auto mapName_noCoplanarFaces = IO::Path("findDragFaces_noCoplanarFaces.map");
+            const auto mapName_twoCoplanarFaces = IO::Path("findDragFaces_twoCoplanarFaces.map");
+
+            auto mapName = GENERATE_COPY(mapName_noCoplanarFaces,
+                                         mapName_twoCoplanarFaces);
+
+            const auto mapPath = IO::Disk::getCurrentWorkingDir() + IO::Path("fixture/test/View/ResizeBrushesToolTest") + mapName;
+            auto [document, game, gameConfig] = View::loadMapDocument(mapPath, "Quake", Model::MapFormat::Valve);
+
+            document->selectAllNodes();
+
+            auto brushes = document->selectedNodes().brushes();
+            REQUIRE(brushes.size() == 2);
+
+            // The two faces of interest
+            const Model::BrushFace& largerTopFace = brushes.at(0)->brush().face(brushes.at(0)->brush().findFace("larger_top_face").value());
+            const Model::BrushFace& smallerTopFace = brushes.at(1)->brush().face(brushes.at(1)->brush().findFace("smaller_top_face").value());
+
+            // Find the entity defining the camera position for our test
+            Model::EntityNode* cameraEntity = kdl::vec_filter(document->selectedNodes().entities(),
+                                                             [](const Model::EntityNode* node){ return node->entity().classname() == "trigger_relay"; }).at(0);
+
+            // Fire a pick ray at largerTopFace
+            const auto pickRay = vm::ray3(cameraEntity->entity().origin(),
+                                          vm::normalize(largerTopFace.center() - cameraEntity->entity().origin()));
+
+            auto tool = ResizeBrushesTool(document);
+
+            Model::PickResult pickResult = Model::PickResult::byDistance(document->editorContext());
+            document->pick(pickRay, pickResult); // populate pickResult
+
+            const Model::Hit hit = tool.pick3D(pickRay, pickResult);
+            CHECK(hit.type() == ResizeBrushesTool::Resize3DHitType);
+            CHECK(!vm::is_nan(hit.hitPoint()));
+
+            const Model::BrushFaceHandle hitTarget = hit.target<Model::BrushFaceHandle>();
+            REQUIRE(hitTarget.face() == largerTopFace);
+            REQUIRE(hit.isMatch());
+            pickResult.addHit(hit);
+
+            // Find the faces that we would drag when pressing Shift
+            REQUIRE(!tool.hasDragFaces());
+            tool.updateDragFaces(pickResult);
+            REQUIRE(tool.hasDragFaces());
+
+            const std::vector<const Model::BrushFace*> dragFaces =
+                kdl::vec_transform(tool.dragFaces(),
+                                   [](const Model::BrushFaceHandle& handle) { return &handle.face(); });
+            
+            if (mapName == mapName_noCoplanarFaces) {
```

Comment:
I'd suggest to set up these expectations along with the input as a table, see https://github.com/TrenchBroom/TrenchBroom/blob/357b2a56ff36585c4192458e0dadd8563028fff4/common/test/src/IO/WorldReaderTest.cpp#L1471 for an example.